packages/core/components/InputOrGroup/styles.module.css

```diff
@@ -33,109 +32,143 @@
     no-repeat;
   background-size: 12px;
   background-position: calc(100% - 12px) calc(50% + 3px);
   background-repeat: no-repeat;
-  background-color: white;
+  background-color: var(--puck-color-white);
+  cursor: pointer;
 }
 
-.Input--readOnly > .Input-input,
-.Input--readOnly > select.Input-input {
-  background-color: var(--puck-color-grey-11);
-  border-color: var(--puck-color-grey-8);
-  color: var(--puck-color-grey-5);
-  opacity: 1;
+@media (hover: hover) and (pointer: fine) {
+  .Input:has( > input):hover .Input-input:not([readonly]),
+  .Input:has( > textarea):hover .Input-input:not([readonly]) {
+    border-color: var(--puck-color-grey-05);
+    transition: none;
+  }
+  .Input:has( > select):hover .Input-input:not([disabled]) {
+    background-color: var(--puck-color-azure-12);
+    background-image: url("data:image/svg+xml;utf8,<svg xmlns='http://www.w3.org/2000/svg' width='100' height='100' fill='%235a5a5a'><polygon points='0,0 100,0 50,50'/></svg>");
+    border-color: var(--puck-color-grey-05);
+    transition: none;
+  }
 }
 
-.Input-input:hover {
-  border-color: var(--puck-color-neutral-3);
+.Input-input:focus {
+  border-color: var(--puck-color-grey-05);
+  outline: 2px solid var(--puck-color-azure-05);
+  outline-offset: 2px;
+  transition: none;
 }
 
-.Input-input:focus {
-  border-color: var(--puck-color-azure-4);
-  outline: var(--puck-color-azure-8) 4px solid;
-  outline-offset: 0;
+.Input--readOnly > .Input-input,
+.Input--readOnly > select.Input-input {
+  background-color: var(--puck-color-grey-10);
+  border-color: var(--puck-color-grey-09);
+  color: var(--puck-color-grey-04);
+  cursor: default;
+  opacity: 1;
+  outline: 0;
+  transition: none;
 }
 
 .Input-radioGroupItems {
   display: flex;
-  border: 1px solid var(--puck-color-grey-7);
+  border: 1px solid var(--puck-color-grey-09);
   border-radius: 4px;
   flex-wrap: wrap;
-  overflow: hidden;
 }
 
 .Input-radio {
-  border-right: 1px solid var(--puck-color-grey-7);
+  border-right: 1px solid var(--puck-color-grey-09);
   flex-grow: 1;
 }
 
+.Input-radio:first-of-type {
+  border-bottom-left-radius: 4px;
+  border-top-left-radius: 4px;
+}
+
 .Input-radio:last-of-type {
-  border-right: none;
+  border-bottom-right-radius: 4px;
+  border-right: 0;
+  border-top-right-radius: 4px;
 }
 
 .Input-radioInner {
-  background-color: white;
-  color: var(--puck-color-grey-4);
+  background-color: var(--puck-color-white);
+  color: var(--puck-color-grey-04);
+  cursor: pointer;
   font-size: var(--puck-font-size-xxxs);
   padding: 8px 12px;
   text-align: center;
+  transition: background-color 50ms ease-in;
 }
 
-.Input-radioInner:hover {
-  background-color: var(--puck-color-azure-9);
-  cursor: pointer;
+.Input-radio:has(:focus-visible) {
+  outline: 2px solid var(--puck-color-azure-05);
+  outline-offset: 2px;
```

Comment:
Done

Reply:
Nice! Much prefer!

I think a `-1px` offset on the inputs would actually be right to account for the border, but we can pick this up later to avoid going back and forth

`-1px` example -

<img width="735" alt="image" src="https://github.com/measuredco/puck/assets/985961/5321692a-9eb4-4f8c-bb87-4a1013cbb840">

Reply:
Also noticing that the offset has been tweaks for the Inputs, but not any of the other focusable elements (external fields, radios)

I guess buttons are more complicated.

Now seeing why you went with `2px` for consistency 🤔 Not quite sure of best approach here - perhaps easier to discuss on a call.

Reply:
I thought you were only refering to the inputs, hence why I only changed those.

Reply:
Yes let's discuss on call.

Reply:
Going to resolve this and merge - can always pick up later if we want